client/src/components/App/Content/InvestigationForm/TabManagement/ContactQuestioning/ContactSection/Schemas/contactQuestioningPersonalSchema.ts

```diff
@@ -1,22 +1,29 @@
 import * as yup from 'yup';
 
-import { invalidPhoneText } from 'commons/Schema/messages';
 import ContactStatusCodes from 'models/enums/ContactStatusCodes';
 import InteractedContactFields from 'models/enums/InteractedContact';
+import { invalidPhoneText, requiredText } from 'commons/Schema/messages';
 import ContactIdValidationSchema from 'Utils/Contacts/ContactIdValidationSchema';
 import { PHONE_NUMBER_REGEX, NOT_REQUIRED_PHONE_NUMBER_REGEX} from 'commons/Regex/Regex';
 
 export const contactQuestioningPersonal = {
     [InteractedContactFields.IDENTIFICATION_NUMBER]: yup
         .string()
         .when(
-            [InteractedContactFields.CONTACT_STATUS, InteractedContactFields.DOES_NEED_ISOLATION],
-            (contactStatus: number, needIsolation: boolean, schema: any, { originalValue }: { originalValue: string }) => {
-                return contactStatus === ContactStatusCodes.COMPLETED || (originalValue === '' && !needIsolation)
+            [InteractedContactFields.CONTACT_STATUS, InteractedContactFields.DOES_NEED_ISOLATION, InteractedContactFields.IDENTIFICATION_TYPE],
+            (contactStatus: number, needIsolation: boolean, identificationType: number, schema: any, { originalValue }: { originalValue: string }) => {
+                return (contactStatus === ContactStatusCodes.COMPLETED ||  !needIsolation) &&  identificationType == null
```

Comment:
why ==?

Reply:
the value of the field changed between null to undefined, ( == null) checks both cases

Reply:
oh, cool